packages/core/src/components/toggletip/toggletip.tsx

```diff
@@ -23,81 +30,100 @@ export class Toggletip {
-      this.close();
-    }
-  };
+  private containerElement?: HTMLDivElement;
+  private tipElementRef: HTMLElement | undefined;
+  private tipArrowElementRef: HTMLElement | undefined;
+  private cleanUp: ReturnType<typeof positionTooltip> | undefined;
 
-  private keyDownListener = (event: KeyboardEvent) => {
-    if (!event.defaultPrevented && event.key === "Escape") {
-      this.close();
-      this.infoButton?.setFocus();
-      event.preventDefault();
-    }
+  disconnectedCallback(): void {
+    this.cleanUp?.();
+    this.cleanUp = undefined;
+  }
 
-    return;
+  private click = (next?: boolean) => {
+    this.active = next !== undefined ? next : !this.active;
   };
 
   render() {
     return (
       <>
-        <dso-info-button
-          aria-describedby="toggle"
-          onClick={this.click}
-          label={this.label}
-          active={this.active}
-          secondary={this.secondary}
-          ref={(element) => (this.infoButton = element)}
-        />
-        <dso-tooltip
-          stateless
-          descriptive
-          id="toggle"
-          strategy="absolute"
-          active={this.active}
-          position={this.position}
+        <div ref={(element) => (this.containerElement = element)}>
+          {["toggle", "secondary"].includes(this.mode) && (
+            <dso-info-button
+              aria-describedby="tooltip"
+              label={this.label}
+              active={this.active}
+              secondary={this.mode === "secondary"}
+              onDsoToggle={({ detail }) => this.click(detail.active)}
+            />
+          )}
+          {this.mode === "badge" && <dso-badge status={this.badgeStatus}>{this.label}</dso-badge>}
+          {this.mode === "icon" && (
+            <dso-icon icon={this.active && this.iconActive ? this.iconActive : this.icon}></dso-icon>
+          )}
+        </div>
+        <Tooltip
           small={this.small}
+          visible={this.visible}
+          onAfterHidden={() => this.tipElementRef?.hidePopover()}
+          tipElementRef={(element) => (this.tipElementRef = element)}
+          tipArrowElementRef={(element) => (this.tipArrowElementRef = element)}
         >
           <slot />
-        </dso-tooltip>
+        </Tooltip>
       </>
     );
   }
+
+  componentDidRender() {
+    if (this.tipElementRef) {
+      const open = this.tipElementRef.matches(":popover-open");
+
+      if (this.active && !open) {
+        this.tipElementRef.showPopover();
+        this.visible = true;
+      } else if (!this.active && open) {
+        this.visible = false;
+      }
+    }
+
+    if (this.active && !this.cleanUp && this.containerElement && this.tipElementRef && this.tipArrowElementRef) {
+      this.cleanUp = positionTooltip(this.containerElement, this.tipElementRef, this.tipArrowElementRef, this.position);
+    } else if (!this.active && this.cleanUp) {
+      this.cleanUp();
+      this.cleanUp = undefined;
```

Comment:
Achteraf gezien is het netter om deze twee regels code in de event handler op regel 100 bij te voegen. Alles wat met opruimen te maken heeft staat dan bij elkaar.

Zullen we er een losse event handler van maken en de logica verplaatsen?

Bijkomend voordeel is dat tijdens het weganimeren de tooltip nog steeds wordt gepositioneerd 🤓